### ISSUE TITLE
Transparent RSyntaxTextArea

### ISSUE THREAD

**Issue by mgarin** [bobbylight/RSyntaxTextArea#53]
Usually it is enough to set `setOpaque ( false )` on any Swing component to disable its background painting but its not the case for RSyntaxTextArea. To fully disable its background you have to provide transparent color in addition to opacity:
```java
syntaxArea.setBackground ( new Color ( 255, 255, 255, 0 ) );
syntaxArea.setOpaque ( false );
```
I guess its just a small issue that background is always filled with the specified background color, but not only when component is opaque. 

Also pretty strange that if I make area non-opaque first and only then apply the backrgound color:
```java
syntaxArea.setOpaque ( false );
syntaxArea.setBackground ( new Color ( 255, 255, 255, 0 ) );
```
I will see something like this:
![image](https://cloud.githubusercontent.com/assets/2361389/3430887/b7985500-0065-11e4-84f7-81a3e7f5789e.png)
With another order everything is fine:
![image](https://cloud.githubusercontent.com/assets/2361389/3430899/d26baff8-0065-11e4-8987-2ebc9a09ed78.png)
This issue might probably have the same roots.

And there is one more funny thing about `setOpaque ( false )` - current line highlight is not painting anymore if I make RSyntaxTextArea non-opaque (even without providing transparent background color). I guess current line highlight is not connected to background and it can also be disabled by a separate call to `setHighlightCurrentLine ( false )` already, so it is a bit pointless to connect opacity with current line painting I guess.

Anyway, its only my thoughts, probably I am missing something :)

**Comment by miho** [comment]
Checkout issue #102 and the related pull request. We had similar issues but where able to fix them.

**Comment by bobbylight** [comment]
This was fixed by commit 280f65d.

You now cannot get rendering garbage by making method calls in the "wrong" order, and extra styles like the current line highlight, bracket matching, etc. are rendered properly even when translucency is enabled.

For the short term, you still have to set the editor to a non-opaque color (alpha < 255) for translucency to be enabled, but note that you *don't* have to explicitly call setOpaque(false) any longer, as the non-opaque color will now be detected and the opaque state toggled automatically.  In a future commit (before the next release) the code for adding images to the background of RSTA will change so it is no longer an either-or thing with the background color; when that commit is made, calling setOpaque(false) by itself will be another means of completely turning off background (color) painting, as is expected.
